marlowe-runtime-web/server/Language/Marlowe/Runtime/Web/Server/REST/Contracts.hs

```diff
@@ -134,9 +133,7 @@ put contractId body = do
 
       tx <- case req of
         Nothing -> throwError $ badRequest' "Invalid text envelope cbor value"
-        Just (Left tx) -> do
-          unless (getTxBody tx == txBody) $ throwError (badRequest' "Provided transaction body differs from the original one")
```

Comment:
Removed this check - the tx body determines the TxId, so it's unnecessary and caused issues with the script validity which could be semantically equal but structurally different (same issue as `CliSpec.hs`)